Compilation error caused by latest change to overlap

I'm getting a compilation error (see below) that's related to the recent change to the overlap detection.
```
ic/core.hpp:46:
In file included from /Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/slice.hpp:5:
In file included from /Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/NoneType.hpp:6:
In file included from /Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/builtins/bool_.hpp:6:
In file included from /Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/tuple.hpp:8:
In file included from /Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/ndarray.hpp:38:
/Users/jlaroche/ua_conda-Darwin.x86_64/miniforge/envs/jean/lib/python3.8/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:58:14: error: no member named 'baseid' in
      '(anonymous namespace)::pythonic::types::sliced_list<long, (anonymous namespace)::pythonic::types::cstride_slice<1>>'
    return e.baseid();
```
I hope this is enough to get an idea of where this happens

--- a/pythran/pythonic/types/numpy_gexpr.hpp
+++ b/pythran/pythonic/types/numpy_gexpr.hpp
@@ -89,6 +89,12 @@ namespace types
     return false;
   }
 
+  template <class E0, class E1, class E2>
+  bool may_overlap(E0 const &e0, types::numpy_vexpr<E1, E2> const &e1)
+  {
+    return may_overlap(e0, e1.data_);
+  }
+
   template <class E0, class Op, class... Args, size_t... Is>
   bool may_overlap(E0 const &e0, types::numpy_expr<Op, Args...> const &e1,
                    utils::index_sequence<Is...>)

--- a/pythran/tests/test_ndarray.py
+++ b/pythran/tests/test_ndarray.py
@@ -715,6 +715,25 @@ def newaxis8(n):
                       30,
                       gexpr_composition22=[int])
 
+    def test_gexpr_vexpr0(self):
+        self.run_test("def gexpr_vexpr0(x, y): x[1:3] = x[y]",
+                      numpy.arange(10), numpy.array([0, 1]),
+                      gexpr_vexpr0=[NDArray[int,:], NDArray[int, :]])
+
+    def test_gexpr_vexpr1(self):
+        self.run_test("def gexpr_vexpr1(x, y): x[1:3] = y[y]",
+                      numpy.arange(10), numpy.array([0, 1]),
+                      gexpr_vexpr1=[NDArray[int,:], NDArray[int, :]])
+
+    def test_gexpr_baseid0(self):
+        self.run_test("def gexpr_baseid0(x, y): x[1:3] = y[0]",
+                      numpy.arange(10), numpy.array([[0, 1]]),
+                      gexpr_baseid0=[NDArray[int,:], NDArray[int, :, :]])
+
+    def test_gexpr_baseid1(self):
+        self.run_test("def gexpr_baseid1(x, y): x[:] = y.T",
+                      numpy.arange(4).reshape(2,2), numpy.array([[0, 1], [2, 3]]),
+                      gexpr_baseid1=[NDArray[int,:,:], NDArray[int, :, :]])
 
     def test_gexpr_copy0(self):
         self.run_test("def gexpr_copy0(a,b): a[:,0] = b[:,0]; return a",

